File: Discord Bot/Bot.lua
local discordia = require('discordia')
local client = discordia.Client()
local prefix = "!"
local file = io.open("token.txt", "rb")
local token = file:read "*a"
local Hero = require("Hero")
file:close()

client:on('messageCreate', function(message)
    local text = string.lower(message.content)

    if string.match(text, "!") then
        text = text:gsub("!", "")
        if string.match(text, "gen") or string.match(text, "generate") then 
            if string.match(text, "all") then
                message.channel:send("Your Hero Is: "..Hero.Gen())
            elseif string.match(text, "tank") then
                message.channel:send("Your Hero Is: "..Hero.GenTank())
            elseif string.match(text, "damage") then
                message.channel:send("Your Hero Is: "..Hero.GenDamage())
            elseif string.match(text, "support") then
                message.channel:send("Your Hero Is: "..Hero.GenSupport())
            else
                message.channel:send("Hero Type Not Found, Hero Types are All, Tank, Damage, and Support")
            end
        end
    end
end)

client:run("Bot "..token)

File: Discord Bot/bot.js
//Vars
const Discord = require('discord.js')
const client = new Discord.Client()
const Hero = require('./hero.js')
var fs = require('fs');

//Core
client.on('ready', () => {
    console.log("Connected as " + client.user.tag)
    client.user.setActivity("!Generate"); 
})

client.on('message', (msg) => {
    if (msg.author != client.user) {
        if  (msg.content.startsWith("!")) {
            ProcessCmd(msg)
        }
    }
})

//Process Command
function ProcessCmd(msg) {
    let fullCommand = msg.content.substr(1)
    let splitCommand = fullCommand.split(" ")
    let primaryCommand = splitCommand[0].toLowerCase()
    let arguments
    if (splitCommand[1]) {
        arguments = splitCommand[1]
    } else {
        arguments = ""
    }
    if (primaryCommand == "gen" || primaryCommand == "generate") {
        GenerateCmd(msg, arguments)
    }
}

//Commands
function GenerateCmd(msg, args) {
    if (args == "all") {
        msg.channel.send("Your Hero Is: " + Hero.Gen())
    } else if (args == "tank") {
        msg.channel.send("Your Hero Is: " + Hero.GenTank())
    } else if (args == "damage") {
        msg.channel.send("Your Hero Is: " + Hero.GenDamage())
    } else if (args == "support") {
        msg.channel.send("Your Hero Is: " + Hero.GenSupport())
    } else {
        msg.channel.send("Hero Type Not Found, Hero Types are All, Tank, Damage, and Support")
    }
}

//Read Token Then Run
fs.readFile('token.txt', 'utf8', function(err, Token) {
    if (err) throw err;
    client.login(Token)
});